pkg/host/host.go

```
@@ -0,0 +1,241 @@
+package host
+
+import (
+	"context"
+	"crypto/rand"
+
+	"github.com/kataras/go-events"
+	"github.com/libp2p/go-libp2p"
+	cmgr "github.com/libp2p/go-libp2p-connmgr"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	dsc "github.com/libp2p/go-libp2p-discovery"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	mplex "github.com/libp2p/go-libp2p-mplex"
+	ps "github.com/libp2p/go-libp2p-pubsub"
+	direct "github.com/libp2p/go-libp2p-webrtc-direct"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/pion/webrtc/v3"
+	"github.com/sonr-io/sonr/pkg/config"
+	types "go.buf.build/grpc/go/sonr-io/motor/core/v1"
+)
+
+// hostImpl type - a p2p host implementing one or more p2p protocols
+type hostImpl struct {
+	// Standard Node Implementation
+	host host.Host
+	SonrHost
```

This line shouldn't be here if you're trying to implement this interface. In fact, it silences any errors that arise from it not being implemented fully.